osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerQueueList.cs

```diff
@@ -40,5 +47,43 @@ protected override void LoadComplete()
 
             public override IEnumerable<Drawable> FlowingChildren => base.FlowingChildren.OfType<RearrangeableListItem<PlaylistItem>>().OrderBy(item => item.Model.PlaylistOrder);
         }
+
+        private class QueuePlaylistItem : DrawableRoomPlaylistItem
+        {
+            public readonly IBindableList<PlaylistItem> Items = new BindableList<PlaylistItem>();
+
+            [Resolved]
+            private IAPIProvider api { get; set; }
+
+            [Resolved]
+            private MultiplayerClient multiplayerClient { get; set; }
+
+            [Resolved(typeof(Room), nameof(Room.QueueMode))]
+            private Bindable<QueueMode> queueMode { get; set; }
+
+            public QueuePlaylistItem(PlaylistItem item)
+                : base(item)
+            {
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                RequestDeletion = item => multiplayerClient.RemovePlaylistItem(item.ID);
+
+                Items.BindCollectionChanged((_, __) => updateDeleteButtonVisibility());
+                queueMode.BindValueChanged(_ => updateDeleteButtonVisibility());
+                SelectedItem.BindValueChanged(_ => updateDeleteButtonVisibility(), true);
+            }
+
+            private void updateDeleteButtonVisibility()
+            {
+                AllowDeletion = queueMode.Value != QueueMode.HostOnly
+                                && Items.Count > 1
+                                && Item.OwnerID == api.LocalUser.Value.OnlineID
+                                && SelectedItem.Value != Item;
```

Comment:
The `Items.Count > 1` check seems unnecessary (as it was removed from server).

(I'm deliberately ignoring the fact that there's no check for `Expired` here, since those items should never be added to this list. They should reside in `MultiplayerHistoryList` instead.)